packages/collector/test/tracing/api/test.js

```diff
@@ -86,18 +87,19 @@ mochaSuiteFn('tracing/api', function () {
           expect(span.handleConstructorName).to.equal('SpanHandle');
         }));
 
-    it('must mark the current span as erroneous', () =>
+    it.only('must mark the current span as erroneous', () =>
       controls
         .sendRequest({
           path: '/span/mark-as-erroneous'
         })
-        .then(response => {
-          const span = response.span;
-          expect(span).to.exist;
-          expect(span.name).to.equal('node.http.server');
-          expect(span.errorCount).to.equal(1);
-          expect(span.data.http.error).to.equal(
-            'This call has been marked as erroneous via the Instana Node.js SDK, no error message has been supplied.'
+        .then(() => {
+          return retry(() =>
+            agentControls.getSpans().then(spans => {
```

Comment:
TBD: The test was not correct. We have compared the response span data. This data is not the same data as the transmitted data. I think this has a bigger underlying issue. We keep the span as instance in the instrumentation.
The span we ask with `cls` is not the same instance as far as I can see.

Reply:
Gnaah, good find. The test is indeed unsuitable :-//  

> The span we ask with cls is not the same instance as far as I can see.

No, I'm pretty sure it is the same instance. I think it is a bit simpler: We basically write `span.ec` at the end of every instrumentation, before we close the span. Like [here](https://github.com/instana/nodejs/blob/main/packages/core/src/tracing/instrumentation/protocols/httpServer.js#L186-L190). 

If I remove line 189 (span.ec = ...), the test passes, even without disableAutoEnd etc. 

We probably could to do something similar as we do with `span.data.http.path_tpl` (see freezePathTemplate method in spanHandle.js): Once `span.ec` has been set explicitly via the SDK, we need to remember that fact and make span.ec read-only.

Reply:
And the reason why the disableAutoEnd/end combo fixes the test is because `span.end()` directly tansmits the span, in this case without overwriting `span.ec`.